profiling/src/profile/mod.rs

```diff
@@ -208,6 +211,7 @@ pub struct EncodedProfile {
     pub start: SystemTime,
     pub end: SystemTime,
     pub buffer: Vec<u8>,
+    pub endpoints_stats: Box<ProfiledEndpointsStats>,
```

Comment:
I don't see the need for Boxing here

Reply:
Actually I could see a use for it if you expect the `ProfiledEndpointsStats` to be empty most of the time.

An empty Hashmap is [48 bytes](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=89500f98178f6ca926bd25e3b887e63f) but if you use `Option<Box<ProfiledEndpointsStats>>`, in case the ProfiledEndpointsStats is empty you can put a None variant and have it take only 8 bytes instead.

Reply:
> I don't see the need for Boxing here since this is not a C struct

I'm not sure I understand this comment.

Reply:
> I'm not sure I understand this comment.

The profiling::EncodedProfile struct is the dual to the the ffi::EncodedProfile one, but that doesn't support repr(C). The `endpoints_stats` field in ffi::EncodedProfile needs to be Boxed, but this one doesn't so you could just have `pub endpoints_stats: ProfiledEndpointsStats` here.

Reply:
✔️